scripts/load-spatial-data.sh

```diff
@@ -0,0 +1,45 @@
+if [ -z "$1" ]; then
+    echo 'Please specify a space to load spatial data into (i.e. beta)' >&2
+    exit 1
+fi
+
+if [ ! $(command -v jq) ] || [ ! $(command -v cf) ]; then
+    echo "jq and cf packages must be installed. Please install via your preferred manager."
+    exit 1
+fi
+
+TARGET="weathergov-$1"
+
+cf target -o nws-weathergov -s $1
+
+# Get an available local port.
+LOCAL_PORT=$(netstat -aln | awk '
+  $6 == "LISTEN" {
+    if ($4 ~ "[.:][0-9]+$") {
+      split($4, a, /[:.]/);
+      port = a[length(a)];
+      p[port] = 1
+    }
+  }
+  END {
+    for (i = 3000; i < 65000 && p[i]; i++){};
+    if (i == 65000) {exit 1};
+    print i
+  }
+')
+
+# Trap exit signals so we can force our background process to exit as well.
+trap "exit" SIGINT SIGTERM
+trap "kill 0" EXIT
+
+# Get database service connection details from cf
+cf curl "/v2/apps/$(cf app --guid $TARGET)/env" | jq -r '.system_env_json.VCAP_SERVICES["aws-rds"][0].credentials' | jq -r '[.host, .port, .db_name, .username, .password]|@tsv' |
```

Comment:
Masterful.

Reply:
I stole this from one of Logan's scripts and just tweaked it a little. It's very nice!